espressif/vscode-esp-idf-extension
# Feature/hints viewer

## Description

Added Hints Viewer. Hints are provided in yaml file from esp-idf folder: <ESP-IDF FOLDER>/tools/idf_py_actions/hints.yml

## Type of change

Please delete options that are not relevant.
- New feature (non-breaking change which adds functionality)

## Steps to test this pull request

Provide a list of steps to test changes in this PR and required output

1. Open ERROR HINTS bottom panel:
![image](https://github.com/espressif/vscode-esp-idf-extension/assets/25748706/9642fab8-2a49-4881-9632-105a9bc44130)

2. Press on "ESP-IDF: Search Error Hint" 
![image](https://github.com/espressif/vscode-esp-idf-extension/assets/25748706/200859cf-47fe-443c-a3a1-499c38a16a6a)

3. Paste error in the top input that shows up
![image](https://github.com/espressif/vscode-esp-idf-extension/assets/25748706/be830e57-8654-4695-bb32-b7452ff20cb8)

- Expected behaviour:
Hints shown in the panel
![image](https://github.com/espressif/vscode-esp-idf-extension/assets/25748706/104757d5-9128-48d0-bb22-8d588720847d)


## How has this been tested?

Followed the steps above on a windows OS.

**Test Configuration**:
* ESP-IDF Version: 5.1.1
* OS (Windows,Linux and macOS): Windows

## Checklist
